spras/config.py

```diff
@@ -225,6 +240,13 @@ def process_config(self, raw_config):
         self.analysis_include_graphspace = raw_config["analysis"]["graphspace"]["include"]
         self.analysis_include_cytoscape = raw_config["analysis"]["cytoscape"]["include"]
         self.analysis_include_ml = raw_config["analysis"]["ml"]["include"]
+        self.analysis_include_evalution = raw_config["analysis"]["evaluation"]["include"]
+
+        # COMMENT: the code will run correctly without this section below due to empty dict in try except above
+        # TODO: decide if this part is needed
+        if self.gold_standards == {} and self.analysis_include_evalution == True:
+            print("Gold standard data not provided. Evaluation analysis cannot run.")
+            self.analysis_include_evalution = False
 
```

Comment:
Do we also need to do error-checking to see if all the dataset labels in the gold standards are existing datasets labels?

Reply:
The code will break in the snakemake rule merge_input with error `InputFunctionException in rule merge_input in file /Users/nehatalluri/Desktop/research/spras/Snakefile, line 152: Error: KeyError: <the dataset that does not exist>`

There error isn't nice, but it does work as expected when the dataset label does not exist in the config file.